### PR TITLE
Add some theory about partitioned sets

### PR DESCRIPTION
**Pull request by xrchz**

### CODE REVIEW

File: src/pred_set/src/pred_setScript.sml [hunk 1]
```diff
@@ -5474,6 +5474,20 @@ val equiv_on_def = new_definition(
        (!x y z. x IN s /\ y IN s /\ z IN s /\ R x y /\ R y z ==> R x z)``);
 val _ = set_fixity "equiv_on" (Infix(NONASSOC, 425))
 
+(* Overload equivalence class of a relation *)
+val _ = overload_on("equiv_class", ``\R s x. {y | y IN s /\ R x y}``);
```
Comment (mn200) [inline src/pred_set/src/pred_setScript.sml:5478]: Please change to use Overload notation

**Comment by mn200** [comment]
Nice!

**Comment by binghe** [comment]
Can we add top-level universal quantifiers to the newly added theorems?

**Comment by xrchz** [comment]
> Can we add top-level universal quantifiers to the newly added theorems?

It's not my preferred style, but I don't much mind either way. @mn200 any opinion?

**Comment by xrchz** [comment]
> > Can we add top-level universal quantifiers to the newly added theorems?
> 
> It's not my preferred style, but I don't much mind either way. @mn200 any opinion?

I've gone ahead and done this.

**Comment by mn200** [comment]
Thanks for this!

**Pull request merged**